src/shell/shell.h

```diff
@@ -58,4 +63,34 @@ int flux_shell_add_event_handler (flux_shell_t *shell,
                                   flux_msg_handler_f cb,
                                   void *arg);
 
+/*  flux_shell_task_t API:
+ */
+
+/*  Return the current task for task_init, task_exec, and task_exit callbacks:
+ *
+ *  Returns NULL in any other context.
+ */
+flux_shell_task_t * flux_shell_current_task (flux_shell_t *shell);
+
+/*
+ *  Return the cmd structure for a shell task.
+ */
+flux_cmd_t *flux_shell_task_cmd (flux_shell_task_t *task);
+
+/*
+ *  Return the subprocess for a shell task in task_fork, task_exit callbacks:
+ */
+flux_subprocess_t *flux_shell_task_subprocess (flux_shell_task_t *task);
+
+/*  Call ready_cb when channel `name` is ready for reading.
```

Comment:
There can't be multiple subscribers to a given channel. The first subscriber wins and subsequent attempts return EEXIST. (In this simplistic approach, it is literally 4 lines or something)

Is "subscribe" the wrong verb here?

Side note: it would be neat in the future if shell plugins could subscribe to output and get passed an "ioencode" object which they could inspect, possibly modify, and pass on the result to the next plugin. Maybe that could be a future interface.

Reply:
Ah OK.  I guess I was confused by the terminology, and maybe by past conversations, but that doesn't necessarily mean there's anything wrong with this.  Maybe just a comment that there can be only one subscriber per channel would be sufficient for now.